### ISSUE TITLE
OSX 10.10.5 - Files locking up on smb connection

### ISSUE THREAD

**Issue by elneco** [atom/atom#9824]
I'm editing files on a mounted volume (connected via SMB). 

When I try to discard changes on a file w/ Git the file locks up and i get a "Permission Denied" message in atom. I can not delete the file in finder as well. 

I then need to disconnect from the mounted volume or reboot to get the file back.

Please help.

**Comment by SamPotts** [comment]
Happens on El Capitan as well. Makes the editor pretty useless for network work as Atom doesn't even unlock the file on close. Best solution would be to not lock the file. Sublime manages it so it must be possible?

**Comment by thedaniel** [comment]
Related: #1766

Does this happen when Atom is started in safe mode as well?

**Comment by SamPotts** [comment]
Yep. Just checked and it still locks files in safe mode.

**Comment by SamPotts** [comment]
Be good to know when this is fixed. For now it's back to Sublime :-1:

**Comment by elneco** [comment]
Ok so my setup is a bit complicated. 

I use a windows vm to handle git stuff as most of the network drives are windows based and for me git seems faster windows to windows. 

My main machine is a mac and this is where I do all of my dev work. On my mac i connect to network drives via smb. 

Discarding changes from the windows vm on files that are open in atom on my mac - locks the file and in some cases I actually can access the file in atom but the file is blank.

Some network drives i connect to are Ubuntu based - in this scenario I perform git operations on my mac as Git operations OSX-to-Ubuntu is nowhere near as slow as OSX-to-windows conenctions. THIS WORKS FINE. I can discard changes and files dont lock up even when they are still open in atom.

When i try Git operations on my mac with files on a windows share - it is VERY slow - but eventually it discards files just fine but i get some .smbdelete files which can not be deleted?

So it seems the main issue is -> discarding changes w/ git on my WINDOWS machine while that file is open in atom on my MAC locks / empties the file. (Sublime Text handles this w/ no problem)

Overall this is a problem I'll be facing when I need to work on windows based network drives.

**Comment by SamPotts** [comment]
Just tried the beta version, also happens with that :(

**Comment by jordandh** [comment]
I am having the same issue connecting to an SMB share. I am running OSX 10.11.3 on a Mid 2012 MacBook Pro Retina. The share is running on Windows in a virtual machine running within VMWare Fusion on the same mac.

**Comment by pi-mont** [comment]
Same here, on Mac OS 10.11.3 and using SMB to modify files on a Windows Machine right next to me.
This is getting beyond frustrating.

**Comment by SamPotts** [comment]
Yep. I'm using it for any locally hosted projects but it'd be nice to be able to use it for everything. Ah well. Sublime it is for now.

**Comment by davejavou** [comment]
Running OS X 10.10.5,
Using SMB to work files on a remote windows machine.
Same problem as reported. Files are locked when Atom touches them. 

1. Open File
2. Make Changes
3. Save file
4. The file is now locked and cannot be altered by atom, the mac os host finder, or the remote windows machine's own explorer or command prompt.
5. Unmount the windows smb volume.
6. The files are now unlocked and able to be altered.

Thank you for taking a look at this bug! Looking forward to using atom once addressed.

**Comment by jtuds** [comment]
Is there any progress with this? Atom locking files makes using mounted drives and Git very difficult

**Comment by Nantiee** [comment]
Same problem here, this causes us big problems.
Atom is a great editor, but this is a good reason to come back to Sublime Text.

**Comment by ryanmaffesolimv** [comment]
This issue still exists as I have the same experience here using windows 10 via parallels on High Sierra. 
Using Atom on my mac to edit files stored on the NTFS drive over smb locks the files and breaks all git functionality.

**Comment by jtuds** [comment]
I also just went to VSCode

**Comment by sampotts** [comment]
All good. Atom has also been marked as stale.

**Comment by jmmanley** [comment]
Still sad that this hasn't been resolved (v 1.45.0, macOS Catalina) 😢 Perhaps interestingly, Atom works fine if I mount my volume via AFP, but freezes up with the same volume via SMB.

**Comment by jq-87** [comment]
Closing this since Atom will be sunset https://github.blog/2022-06-08-sunsetting-atom/
